src/backend/joanie/signature/backends/lex_persona/client.py

```diff
@@ -0,0 +1,400 @@
+            concerned.
+        """
+        try:
+            url = f"{self.base_url}/api/workflows/{self.workflow_id}/invite"
+            payload = {"recipientEmail": recipient_email}
+            response = requests.post(url, headers=self.headers, json=payload, timeout=3)
+            if response.status_code == 404:
+                raise ValueError(
+                    str(
+                        f"{recipient_email} has not document registered to sign at the moment."
+                    )
+                )
+            return response.json()
+        except ValueError as exception:
+            logger.error("{%s}", exception)
+            return None
+
+    def _retrieve_all_files_to_sign(self, recipient_email: str):
+        """
+        For organization's owners : in order to get all
+        the workflows that are ongoing and the ones that
+        are completed.
+
+        The advantage of this endpoint : it will regroup
+        every workflows that is linked to the recipient
+        email address.
+
+        The API filters them out according to the given
+        email value.
+        """
+        url = f"{self.base_url}/api/workflows/"
+        params = {
+            "item.currentRecipientEmails": recipient_email,
+        }
+        response = requests.get(url, headers=self.headers, params=params, timeout=3)
+        return response.json()
+
+    def _sign_specific_selection_of_workflow(self, workflow_ids: list, token_jwt: str):
+        """
+        Sign a selection of document. This endpoint generates a link
+        where we can go sign directly the document.
+        Note :
+            - payload : {"workflows" : ["workflow_id#1", "workflow_id#2"]}
+        """
+        headers = {
+            "Authorization": f"Bearer {token_jwt}",
+            "Content-Type": "application/json",
+        }
+        url = f"{self.base_url}/api/requests/"
+        payload = {"workflows": workflow_ids}
+        response = requests.post(url, headers=headers, json=payload, timeout=3)
+        return response.json()
+
+    def _verify_webhook_event(self, webhook_event_id: str):
+        """
+        Verify if the webhook event is real.
+
+        As we cannot trust the origin of the request, we should
+        retrieve the webhook event and make sure it exists
+        with this endpoint before handling the notification
+        else we should not update our objects.
+        """
+        url = f"{self.base_url}/api/webhookEvents/{webhook_event_id}"
+        response = requests.get(url, headers=self.headers, timeout=3)
+        if response.status_code == 404:
+            return Response(
+                {
+                    "error": f"Failed, webhook event does not exists : '{webhook_event_id}'"
+                },
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        return Response(
+            {"status": f"Success, webhook event exists : '{webhook_event_id}'"},
+            status=status.HTTP_200_OK,
+        )
```

Comment:
Response may 500 or other, be sure to return a 200 only if response.status_code is in the 200 family